day10/kevin/index.js

```diff
@@ -0,0 +1,44 @@
+const select = (selector) => document.querySelector(selector);
+const checkboxSelector = 'input[type="checkbox"]';
+
+function holdShiftToCheckCheckboxes(selector = '#checklist') {
+  const form = select(selector);
+  const checkboxes = [...form.querySelectorAll(checkboxSelector)];
+  let prevCheckedIndex = null;
+  let isPressShift = false;
+
+  function checkInRange(from = 0, to = 0) {
+    const minOne = Math.min(from, to);
+    const maxOne = Math.max(from, to);
+    checkboxes.forEach((checkbox, i) => {
+      if (i >= minOne && i <= maxOne) checkbox.checked = true;
+    });
+  }
+
+  function findCheckboxIndex(currentCheckbox) {
+    return checkboxes.findIndex((checkbox) => checkbox === currentCheckbox);
+  }
+
+  function toggleIsPressShift(forceState) {
+    isPressShift = !!forceState ?? !isPressShift;
```

Comment:
這個判斷是我有點看無...
如果 `forceState`：
傳 `true`/`false` 進來會經由 `!!` 轉成 `true`/`false`
傳 `null`/`undefined` 就會回 `!isPressShift`

流程是醬子嗎？

Reply:
其實... 我好像寫錯了XDD
似乎怎麼樣都不會走到後面的 `isPressShift`

因為假設是 `undefined`，`!!undefined` 就會被轉成 false，所以不管有沒有傳參數 或 不管傳甚麼
`!!forceState` 一定會是 Boolean

你說的沒錯XD 只是第二句錯誤
傳 `null`/`undefined` 會回傳 false 並附值給 `isPressShift`

Reply:
1. 
> 因為假設是 `undefined`，`!!undefined` 就會被轉成 false，所以不管有沒有傳參數 或 不管傳甚麼 `!!forceState` 一定會是 Boolean

我看懂了！因為 `undefined` 是 falsy value 所以會被轉成 false

2.
> 你說的沒錯XD 只是第二句錯誤 傳 `null`/`undefined` 會回傳 false 並附值給 `isPressShift`

罷特第二句錯誤這個我就不懂了QQ，因為 MDN 的說明是：

> The nullish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is null or undefined, and otherwise returns its left-hand side operand.

我的理解是當左手邊的運算數如果是 `null` or `undefined` 會 return 右手邊的運算數，不然就 return 左手邊的運算數。所以我才會以為是如果 `!!forceState` 是 `null`/`undefined` 就回傳 `isPressShift`，不過經過你第二段的回覆之後我能理解為什麼傳 `null`/`undefined` 會回傳 false 並賦值給 `isPressShift`，那的確好像不會走到後面的 `isPressShift`

然後我好奇為什麼當初會想要使用 `??` 來做判斷～～

Reply:
其實原本的目的是回傳結果一定要是 Boolean 
然後如果沒有傳參數的話回傳 `!isPressShift`，所以那個 `??` 的用意是在沒有傳參數的情況下回傳後面的 `!isPressShift`，來達到 toggle 的功能

所以比較正確的寫法應該是下面這樣，抱歉讓你疑惑了ＸＤ
原本的寫法是錯的

```suggestion
    isPressShift = !! (forceState ?? !isPressShift );
```

Reply:
哦哦哦哦哦！原來如此！下面這樣寫我懂惹！
難怪要用 `!!` 來判斷
感謝凱文的說明～